checker.cpp

```diff
@@ -1,17 +1,27 @@
 #include <assert.h>
 
+bool is_vital_normal(float value, int upperLimit, int lowerLimit) {    // (normalRange = [ 0 , upperLimit-lowerLimit]
+  return ((unsigned)(upperLimit - value) <= (unsigned)(upperLimit - lowerLimit));
+}
+
+//bpm [70,150], spo2[90,100], respRate[30, 95]
 bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
-    return false;
-  } else if(spo2 < 90) {
-    return false;
-  } else if(respRate < 30 || respRate > 95) {
-    return false;
-  }
-  return true;
+  return ( is_vital_normal(bpm,150,70) && is_vital_normal(spo2,100,90) && is_vital_normal(respRate,95,30) );
 }
 
 int main() {
-  assert(vitalsAreOk(80, 95, 60) == true);
-  assert(vitalsAreOk(60, 90, 40) == false);
+  /*
+  //vitalsAreOk(bpm,spo2,respRate)
+  assert(vitalsAreOk(80, 95, 60) == true);        // All Normal
+  assert(vitalsAreOk(60, 95, 60) == false);       //AL_bpm, N_sp02, N_resp
+  assert(vitalsAreOk(180, 95, 60) == false);      //AH_bpm, N_sp02, N_resp
+  assert(vitalsAreOk(80, 95, 22) == false);       //N_bpm,  N_sp02, AL_resp
+  assert(vitalsAreOk(80, 95, 98) == false);       //N_bpm,  N_spo2, AH_resp
+  assert(vitalsAreOk(80, 89, 60) == false);       //N_bpm,  A_sp02, N_resp
+  assert(vitalsAreOk(60, 89, 22) == false);       //All Abnormal Low
+  assert(vitalsAreOk(180, 89, 98) == false);      //All Abnormal High
+  */
+  assert(is_vital_normal(40, 100, 50) == false);
+  assert(is_vital_normal(120, 100, 50) == false);
+  assert(is_vital_normal(92, 100, 50) == true);
```

Comment:
Good test of the abstraction! It is still necessary to test vitalsAreOk, just to ensure that it has covered all the vitals. So it would be sufficient to cover one out-of-range condition per parameter.
Also this will be clearer when the alert (SMS or Sound) is added.